File: app/src/main/java/com/alexkn/syntact/presentation/createbucket/CreateBucketViewModel.java
package com.alexkn.syntact.presentation.createbucket;

import android.os.AsyncTask;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.alexkn.syntact.domain.usecase.bucket.BucketRepository;
import com.alexkn.syntact.restservice.Template;

import org.apache.commons.lang3.StringUtils;

import java.util.Arrays;
import java.util.List;
import java.util.Locale;

import javax.inject.Inject;

public class CreateBucketViewModel extends ViewModel {

    private final List<Locale> availableBuckets;

    private final LiveData<List<Template>> availableTemplates;

    private BucketRepository bucketRepository;

    @Inject
    public CreateBucketViewModel(BucketRepository bucketRepository) {

        super();
        this.bucketRepository = bucketRepository;

        availableBuckets = bucketRepository.getAvailableLanguages();
        availableTemplates = bucketRepository.findAvailableTemplates();
    }

    void addBucket(Locale language, Template template, String words) {

        if (words.isEmpty()) {//TODO
            AsyncTask.execute(() -> bucketRepository
                    .addBucketWithExistingTemplate(language, template == null ? availableTemplates.getValue().get(0) : template));
        } else {
            List<String> phrases = Arrays.asList(StringUtils.split(words, " "));
            AsyncTask.execute(() -> bucketRepository.addBucketWithNewTemplate("AndroidTest", language, phrases));
        }
    }

    public List<Locale> getAvailableBuckets() {

        return availableBuckets;
    }

    public LiveData<List<Template>> getAvailableTemplates() {

        return availableTemplates;
    }
}

File: app/src/main/java/com/alexkn/syntact/presentation/createbucket/CreateBucketViewModel.kt
package com.alexkn.syntact.presentation.createbucket

import android.util.Log
import androidx.lifecycle.*
import com.alexkn.syntact.app.TAG
import com.alexkn.syntact.data.model.Template
import com.alexkn.syntact.core.repository.BucketRepository
import com.alexkn.syntact.core.repository.TemplateRepository
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import java.util.*
import javax.inject.Inject

class CreateBucketViewModel @Inject
constructor(
        private val bucketRepository: BucketRepository,
        private val templateRepository: TemplateRepository
) : ViewModel() {

    var availableLanguages: List<Locale> = bucketRepository.availableLanguages.filter { locale -> locale.language != Locale.getDefault().language }


    var availableTemplates: LiveData<List<Template>> = templateRepository.findTemplates()


    init {
        viewModelScope.launch {
            try {
                templateRepository.updateTemplates()
            } catch (e: Exception) {
                Log.e(TAG, "Update templates: ", e)
            }
        }
    }

    fun addBucketFromExistingTemplate(template: Template) {
        GlobalScope.launch(Dispatchers.Default) {
            bucketRepository.addBucketWithExistingTemplate(template)
        }
    }

}
